port/integration/schema.go

```diff
@@ -0,0 +1,116 @@
+package integration
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+func IntegrationSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Computed: true,
+		},
+		"installation_id": schema.StringAttribute{
+			Required: true,
+		},
+		"version": schema.StringAttribute{
+			Optional: true,
+		},
+		"title": schema.StringAttribute{
+			Optional: true,
+		},
+		"installation_app_type": schema.StringAttribute{
+			Required:   true,
+			Validators: []validator.String{stringvalidator.OneOfCaseInsensitive("WEBHOOK", "KAFKA")},
+		},
```

Comment:
by default users don't know of it, and actually I am less familiar with that as well, therefore I wonder if we want to have it optional with default.

Reply:
This is required in the schema, what would be the default then?

Reply:
It's actually not required by the API, making optional